### PR TITLE
replace black/white language with allow/block

### PR DESCRIPTION
**Pull request by matwerber1**
*Issue #, if available:* /N/A

*Description of changes:*

Searched for terms `black` and `white` and replaced them with block and allow, e.g. `whitelist` becomes `allowlist`, and `blacklist` becomes `blocklist`. Contributing as part of the broader dev community effort to use more inclusive terminology.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by edwardfoyle — APPROVED** [review]

File: packages/amplify-category-auth/provider-utils/awscloudformation/triggers/PreSignup/cloudformation-templates/PreSignup.json.ejs [hunk 1]
```diff
@@ -102,11 +102,11 @@
                     "REGION": { 
                         "Ref": "AWS::Region"
                     },
-                    "DOMAINWHITELIST": {
-                        "Ref": "DOMAINWHITELIST"
+                    "DOMAINALLOWLIST": {
```
Comment (edwardfoyle) [inline packages/amplify-category-auth/provider-utils/awscloudformation/triggers/PreSignup/cloudformation-templates/PreSignup.json.ejs:105]: I think we'll need to be careful with these env variables. Changing them could break existing lambda triggers

**Review by edwardfoyle — CHANGES_REQUESTED** [review]
All but one of these changes looks fine

**Comment by matwerber1** [comment]
@edwardfoyle I used the search to look for instances of `DOMAINWHITELIST` and found three: 
https://github.com/aws-amplify/amplify-cli/search?q=DOMAINWHITELIST

All three are addressed by this PR, would that work? Or, are you perhaps saying this might break pre-existing Amplify projects if someone were to update the CLI and then push changes that use the new code? If so, that's a fair point. 

I'm not deep with Amplify, not sure of the best way to address. Do you have a specific ask of me to help explore? Should this PR stay open? The changes are good ones to make, but I understand that - at least broadly speaking - this is quite the long-term project for the dev community.

**Comment by edwardfoyle** [comment]
@matwerber1 yeah all of the changes here look good except for the changes in `PreSignup.json.ejs`. If an existing project has a pre-signup trigger that uses the blacklist/whitelist environment variables, renaming them to denylist/allowlist will break those existing lambdas.

I think it path forward here would be to update this PR to remove the changes to that file and publish another PR that has those changes behind a [feature flag](https://docs.amplify.aws/cli/reference/feature-flags)

**Comment by attilah** [comment]
Closing in favor of #6331

**Pull request closed without merging**